File: Warmup/ DiagonalDifference.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int a[][] = new int[n][n];
        int pSum = 0, sSum = 0;
        for(int a_i=0; a_i < n; a_i++){
            for(int a_j=0; a_j < n; a_j++){
                a[a_i][a_j] = in.nextInt();
            }
            pSum+=a[a_i][a_i];
            sSum+=a[a_i][n-1-a_i];
        }
        System.out.println(Math.abs(pSum-sSum));
    }
}

File: Warmup/ DiagonalDifference.rb
#!/bin/ruby

n = gets.strip.to_i
a = Array.new(n)
pSum=0
sSum=0
for a_i in (0..n-1)
    a_t = gets.strip
    a[a_i] = a_t.split(' ').map(&:to_i)
    pSum+=a[a_i][a_i]
    sSum+=a[a_i][n-1-a_i]
end
puts (pSum-sSum).abs